nsxt/resource_nsxt_policy_domain.go

```diff
@@ -0,0 +1,336 @@
+}
+
+func createChildDomainDeploymentMap(m interface{}, domainID string, location string) (*data.StructValue, error) {
+	converter := bindings.NewTypeConverter()
+	converter.SetMode(bindings.REST)
+
+	mapID := domainID + "-" + location
+	path := getGlobalPolicyEnforcementPointPathWithLocation(m, location)
+
+	Type := "DomainDeploymentMap"
+	mapObj := model.DomainDeploymentMap{
+		Id:                   &mapID,
+		DisplayName:          &location,
+		EnforcementPointPath: &path,
+		ResourceType:         &Type,
+	}
+
+	childMap := model.ChildDomainDeploymentMap{
+		Id:                  &mapID,
+		ResourceType:        "ChildDomainDeploymentMap",
+		DomainDeploymentMap: &mapObj,
+	}
+
+	dataValue, errors := converter.ConvertToVapi(childMap, model.ChildDomainDeploymentMapBindingType())
+	if len(errors) > 0 {
+		return nil, errors[0]
+	}
+	return dataValue.(*data.StructValue), nil
+}
+
+func setDomainStructWithChildren(m interface{}, domain *model.Domain, locations []string, isUpdate bool) error {
+	var children []*data.StructValue
+
+	// Loop over locations
+	for _, location := range locations {
+		child, err := createChildDomainDeploymentMap(m, *domain.Id, location)
+		if err != nil {
+			return err
+		}
+		if child != nil {
+			children = append(children, child)
+		}
+	}
+
+	if isUpdate {
+		// Delete old existing locations
+		connector := getPolicyConnector(m)
+		converter := bindings.NewTypeConverter()
+		dmClient := gm_domain.NewDefaultDomainDeploymentMapsClient(connector)
+		// Get all currenlt locations
+		objList, err := dmClient.List(*domain.Id, nil, nil, nil, nil, nil, nil)
+		if err != nil {
+			return handleListError("Domain", err)
+		}
+		for _, objInList := range objList.Results {
+			found := false
+			for _, newLocation := range locations {
+				if newLocation == *objInList.DisplayName {
+					found = true
+					break
+				}
+			}
+			if !found {
+				// Remove the unused deployment map
+				toDelete := true
+				lmObj, err2 := convertModelBindingType(objInList, gm_model.DomainDeploymentMapBindingType(), model.DomainDeploymentMapBindingType())
+				if err2 != nil {
+					return err2
+				}
+				deleteObj := lmObj.(model.DomainDeploymentMap)
+				childMap := model.ChildDomainDeploymentMap{
+					Id:                  objInList.Id,
+					ResourceType:        "ChildDomainDeploymentMap",
+					MarkedForDelete:     &toDelete,
+					DomainDeploymentMap: &deleteObj,
+				}
+
+				dataValue, errors := converter.ConvertToVapi(childMap, model.ChildDomainDeploymentMapBindingType())
+				if len(errors) > 0 {
+					return errors[0]
+				}
+				children = append(children, dataValue.(*data.StructValue))
+			}
+		}
+	}
+	domain.Children = children
+	return nil
+}
+
+func resourceNsxtPolicyDomainCreate(d *schema.ResourceData, m interface{}) error {
+	// Initialize resource Id and verify this ID is not yet used
+	id, err := getOrGenerateID(d, m, resourceNsxtPolicyDomainExists)
+	if err != nil {
+		return err
+	}
+
+	if !isPolicyGlobalManager(m) {
```

Comment:
nit: this is already covered in getOrGenerateID

Reply:
Removed